### PR TITLE
Document Binary Search Tree Traversals (+ fix typo)

### LINKED ISSUE #73: Briefly document BST traversal algorithms
See `src/datastructures/BST/README.md`

### PR DESCRIPTION
**Pull request by SteveStoffer**
Hopefully explained the BST Traversals well enough. 

Also while I was reading the README for this the other day, I noticed that `binary_search_tree<T>::inorder_successor(TreeNode<T> *elem, TreeNode<T> *root);` was listed twice. The second one was meant to be inorder_predecessor.

---
### Edit
Closes #73

### CODE REVIEW

**Comment by domfarolino** [comment]
This is to close https://github.com/domfarolino/algorithms/issues/73 right?

File: src/datastructures/binary_search_tree/README.md [hunk 1]
```diff
@@ -611,15 +616,44 @@ to gain even more insight into why our algorithm works correctly, even with dupl
...
+<a name="inorder"></a>
+### `std::vector<T> binary_search_tree<T>::inorder();`
+This method delegates to the `inorder_helper()` method below. This method returns a vector with the elements from the BST in order.
+
+### `void binary_search_tree<T>::inorder_helper(TreeNode<T>* root, std::vector<T> &vec);`
+In order to be able to insert the elements of a BST into a vector, in order, we first need to traverse down the root's left subtree.
+
+ The `inorder_helper(root->left, vec)` method is used to traverse as far down the left subtree as we can go. Once we end up with a node with no left child, we add the value to the vector with `vec.push_back(root->val)`.
+
+ After that we simply use `inorder_helper(root->right)` to traverse down the right subtree, searching for another left subtree.
```
[the remark this answers is not in the document]
Comment (SteveStoffer) [reply src/datastructures/binary_search_tree/README.md:637]: That makes more sense!

File: src/datastructures/binary_search_tree/README.md [hunk 2]
```diff
@@ -611,15 +616,44 @@ to gain even more insight into why our algorithm works correctly, even with dupl
...
+
+<a name="preorder"></a>
+### `std::vector<T> binary_search_tree<T>::preorder();`
+This method delegates to the `preorder_helper()` method below. This method returns a vector with the elements from the BST in preorder.
+
+### `void binary_search_tree<T>::preorder_helper(TreeNode<T>* root, std::vector<T> &vec);`
+The order this method works is as follows:
+- Add the root's value to the vector.
+- Recursively traverse the left subtree.
+- Recursively traverse the right subtree.
```
[the remark this answers is not in the document]
Comment (SteveStoffer) [reply src/datastructures/binary_search_tree/README.md:647]: Can do, this helps visualize what is going on.

**Comment by SteveStoffer** [comment]
> This is to close #73 right?

Yes. Everything should be fixed now... And I guess I should add the issue into the PR comment when I open it.

**Comment by domfarolino** [comment]
Yep, if you edit the OP now it will work too. See https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword.

**Comment by SteveStoffer** [comment]
Cool! Thanks for the link.

File: src/datastructures/binary_search_tree/README.md [hunk 3]
```diff
@@ -611,15 +616,74 @@ to gain even more insight into why our algorithm works correctly, even with dupl
...
+Then we add the first element of the queue into the vector and dequeue it from the queue. The while
+loop will run until the queue is empty, which means each node from the tree was inserted into the
+vector.
+
+<a name="inorder"></a>
+### `std::vector<T> binary_search_tree<T>::inorder();`
+This method delegates to the `inorder_helper()` method below. This method returns a vector with the
+elements from the BST in order.
+
+### `void binary_search_tree<T>::inorder_helper(TreeNode<T>* root, std::vector<T> &vec);`
```
Comment (domfarolino) [inline src/datastructures/binary_search_tree/README.md:639]: Everything here looks good, let's just move the `&` before `vec` in all of these three method headers to right after the `vector<T>` and then I think we're good to go.

**Review by domfarolino — COMMENTED** [review]
Looking good!

**Review by domfarolino — APPROVED** [review]
LGTM! Just remember to clear the commit description box thingy upon merging.

**Pull request merged**